tchannel-core/src/main/java/com/uber/tchannel/handlers/RequestRouter.java

```diff
@@ -206,6 +216,37 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         }
     }
 
+    private ListenableFuture<Response> sendRequestToAsyncHandler(
+            final AsyncRequestHandler asyncHandler, final Request request) {
+        // span used to trace this request
+        final Span span = Tracing.startInboundSpan(request,
+            topChannel.getTracer(), topChannel.getTracingContext());
```

Comment:
@vprithvi you are correct. The way it was handled in in the earlier diff is here: https://github.com/uber/tchannel-java/commit/824cf0937f920e7e6cc6e74118f1c5d98c315b89#diff-0c30656be54ed88283a28294e0fa1555R223

There has been a recent commit in opentracing-java that extracted Noop tracer/span into a reusable module. We could refactor tracing logic to return that noop span when tchnanel is not initialized with it, and not propagate that span at all, only use it as local variable here.  But that's better done as a separate PR. Here the best (although dirty) way is to check for span==null, with a TODO to fix later.